### ISSUE TITLE
Redundant sex column

### ISSUE THREAD

**Issue by eggertkarl** [tutor-web/twstats#5]
Some tables have only the value "T" (presumably for total) in the sex column, see e.g.
> twstats_get_table("eurostat/tespm151/area:EA")$data()

Note: I found that even tables that have only "F" and "M" sexes that the "sex" factor variable was coded as having three levels, "F", "M", and "T". These tables are fine to use. This means that a potential fix would be to exclude tables that actually *contain* the value "T".

**Comment by lentinj** [comment]
Eurostat's vocabulary for sex columns contains the following

```
> get_eurostat_dic('sex')
# A tibble: 7 x 2
  code_name full_name
  <chr>     <chr>
1 T         Total
2 M         Males
3 F         Females
4 DIFF      Absolute difference between males and females
5 NAP       Not applicable
6 NRP       No response
7 UNK       Unknown
```

...currently any ``T``, ``DIFF`` and ``NAP`` are filtered out to simplify the table, but the levels don't get altered. Agreed it would be good to remove the redundant levels also.

**Comment by lentinj** [comment]
Ah, I've missed the point of this. It's that there's a sex column with *just* T, not that the T level is still about.

Generally columns with only one level are thrown away, but because of the special case for sex as above this wasn't happening. I'll rejig this so in the above example there's no sex column.

**Comment by lentinj** [comment]
The above removes loads of redundant sex columns, should fix this.

**Comment by eggertkarl** [comment]
Ah, that's awesome!

I've been sick and stuck in bed for the past few days (hence why there's been very little progress on my end).
I'm starting to feel better though, and look forward to (hopefully) trying it out on Monday. :)

Cheers,
-Eggert Karl Hafsteinsson

________________________________

**Comment by eggertkarl** [comment]
Hey there,

Couldn't sleep (being sick can sure mess up one's sleep cycle...).

I've gone backwards and forwards on the rounding stuff, and I think I've got it sorted (or at least as much as I think is reasonable).

I'll outline the finer details later, but the purpose of this mail is that I ran into a problem when I wanted to test it on the year/value datasets.

First off, the source of this problem may be that I haven't updated material-bank since Wednesday, as I haven't set up git, gitlab keys, etc. on my home machine. So if that's the case, I'm sorry to bother you! :(

The error is that I can't seem to fetch datasets anymore with the twstats library. Searching for datasets works, as does fetching the metadata for them, but running the data() function results in:

twstats_get_table("cbso/82610ENG/total")$data()
Error in gzfile(file, mode) : cannot open the connection
In addition: Warning message:
In gzfile(file, mode) :
  cannot open compressed file 'C:\Users\lenovo\AppData\Local\Temp\RtmpAHZosl\file25b43d6f2949.twstats_cbso/9dce84c5b0a99485', probable reason 'No such file or directory'

(and yes, the 'user' on this laptop is 'lenovo', as I bought it used and it's a real hassle to change that in Windows 10)

I tried using ``devtools::install_github('tutor-web/twstats')`` in case you updated the library itself, but that didn't seem to change anything.

Any ideas?

________________________________
